raiden/network/proxies/token_network.py

```diff
@@ -2213,47 +2239,66 @@ def settle(
             # gas will stop us from sending a transaction that will fail
             pass
 
-        with self.channel_operations_lock[partner]:
-            error_prefix = "Call to settle will fail"
-            gas_limit = self.proxy.estimate_gas(
-                checking_block, "settleChannel", channel_identifier=channel_identifier, **kwargs
-            )
+        if self_or_lc_address == self.node_address:
+            # is not lc
+            with self.channel_operations_lock[partner]:
+                error_prefix = "Call to settle will fail"
+                gas_limit = self.proxy.estimate_gas(
+                    checking_block, "settleChannel", channel_identifier=channel_identifier, **kwargs
+                )
 
-            if gas_limit:
-                error_prefix = "settle call failed"
-                gas_limit = safe_gas_limit(gas_limit, GAS_REQUIRED_FOR_SETTLE_CHANNEL)
+                if gas_limit:
+                    error_prefix = "settle call failed"
+                    gas_limit = safe_gas_limit(gas_limit, GAS_REQUIRED_FOR_SETTLE_CHANNEL)
 
-                transaction_hash = self.proxy.transact(
-                    "settleChannel", gas_limit, channel_identifier=channel_identifier, **kwargs
-                )
-                self.client.poll(transaction_hash)
-                receipt_or_none = check_transaction_threw(self.client, transaction_hash)
+                    transaction_hash = self.proxy.transact(
+                        "settleChannel", gas_limit, channel_identifier=channel_identifier, **kwargs
+                    )
+                    self.client.poll(transaction_hash)
+                    receipt_or_none = check_transaction_threw(self.client, transaction_hash)
 
-        transaction_executed = gas_limit is not None
-        if not transaction_executed or receipt_or_none:
-            if transaction_executed:
-                block = receipt_or_none["blockNumber"]
-            else:
-                block = checking_block
+            transaction_executed = gas_limit is not None
+            if not transaction_executed or receipt_or_none:
```

Comment:
it's a bit hard to understand semantically what _if not transaction_executed or receipt_or_none_ means.

the problem here is the name of the `receipt_or_none` field. please name it to something more descriptive or declare and use a boolean variable to make this `if` more readable